functional/io.py

```diff
@@ -24,12 +25,12 @@ class ReusableFile(Generic[_FileConv_co]):
     def __init__(
         self,
         path: str,
-        delimiter: Optional[str] = None,
+        delimiter: str | None = None,
         mode: str = "r",
         buffering: int = -1,
-        encoding: Optional[str] = None,
-        errors: Optional[str] = None,
-        newline: Optional[str] = None,
+        encoding: str | None = None,
```

Comment:
Python supports the pipe symbol now "I"
So, Optional[str] and str | None are pretty much the same as far as I know

Reply:
Correct

Reply:
So there isn't any point in changing these parts, is there? Unless I am missing something

Reply:
They are identical for typing purposes as far as I know people gravitate to the pipe version due to it being shorter and more concise. Reason I change existing ones was because I ran ruff on this code with the following rule. https://docs.astral.sh/ruff/rules/non-pep604-annotation/

If allowed I am planning to introduce ruff to this project in the future since I have found it very helpful in my personal projects.

Reply:
In general, I prefer the `str | None`, but one argument in favor of the other style is that the pipe is only available in python 3.10 and above. I'm personally fine with dropping support for python 3.8/3.9 in favor of better syntax, but would be accepting of the old syntax if its a substantial barrier to current use.

Reply:
No need to drop support for 3.8/3.9. With `from __future__ import annotations` it allows us to use the new syntax. https://peps.python.org/pep-0563/